### PR TITLE
Events directory restructuring

### PR DESCRIPTION
**Pull request by gpsimenos**
# Summary of changes 
`mbed-os/events` directory restructured according to the internal proposal.

- Public headers moved into `mbed-os/events/include/events` folder
- Unity tests moved into `mbed-os/events/tests/UNITTESTS` folder
- Greentea tests moved into `mbed-os/events/tests/TESTS` folder
- 'Local' unit tests moved into `mbed-os/events/tests/unit` folder
- Include paths updated

This is one of a series of PRs aiming to clean up the mbed-os directory structure. The intention is to create a consistent tree among mbed components, following the below structure:

```
[component name]
├── mbed_lib.json                     // Each component comes with their own config that's appended to the App/System ├── config 
├── CMakeList.txt                     // This depends on the build system choices, part of another PREQ
├── README.md
├── include                           // The top-level include path
│   └── [component name]              // Public headers that can be accessed by the developers - `#include [component name]/*.h`
│       └── internal                  // Internal headers that implement or are included by a public header or are needed by other Mbed OS libraries - `#include [component name]/internal/*.h`
├── source                            // Source files and private headers, available only inside the component's source directory
└── tests
    ├── <framework>                   // <framework> test folder (for example CPPUTEST)
    │   └── <test suite one>          // Can be either source file or a directory
    ├── UNITTESTS                     // Unit tests (previously in UNITTESTS) folder using googletest framework (to be renamed to gtest at a later stage)
    │   └── <unit test suite one>     // Can be either source file or a directory
    └── TESTS                         // Greentea tests (previously in TESTS) folder (to be renamed to greentea at a later stage)
        └── <greentea test suite one> // Can be either source file or a directory
 
mbed-os
└── TESTS (to be renamed to greentea at a later stage)
    ├── integration   
    └── system
```

#### Impact of changes 

None

#### Migration actions required 

None

----------------------------------------------------------------------------------------------------------------

### Documentation 

None

----------------------------------------------------------------------------------------------------------------
### Pull request type 
    [X] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results 
    [] No Tests required for this change (E.g docs only update)
    [X] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR

```
(equeue main.cpp modified to remove MBED_EXTENDED_TESTS check)
mbed test -m K64F -t GCC_ARM --greentea -n events-tests-tests-events-timing,events-tests-tests-events-equeue,events-tests-tests-events-queue

Build successes:
  * K64F::GCC_ARM::EVENTS-TESTS-TESTS-EVENTS-EQUEUE
  * K64F::GCC_ARM::EVENTS-TESTS-TESTS-EVENTS-QUEUE
  * K64F::GCC_ARM::EVENTS-TESTS-TESTS-EVENTS-TIMING
  * K64F::GCC_ARM::MBED-BUILD

===================================================================

(equeue main.cpp modified to remove MBED_EXTENDED_TESTS check)
mbed test -m K64F -t GCC_ARM --greentea -n events-tests-tests-events-timing,events-tests-tests-events-equeue,events-tests-tests-events-queue --app-config TESTS/configs/baremetal.json

Build successes:
  * K64F::GCC_ARM::EVENTS-TESTS-TESTS-EVENTS-EQUEUE
  * K64F::GCC_ARM::EVENTS-TESTS-TESTS-EVENTS-QUEUE
  * K64F::GCC_ARM::EVENTS-TESTS-TESTS-EVENTS-TIMING
  * K64F::GCC_ARM::MBED-BUILD

===================================================================

mbed test --unittests --compile

. . .
[ 11%] Built target events-equeue.MbedOS
. . .
[ 62%] Built target events-equeue
. . .

Unit tests built successfully.
```
    
----------------------------------------------------------------------------------------------------------------
### Reviewers 
@evedon
@ashok-rao
@LDong-Arm

----------------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by LDong-Arm** [comment]
Thanks @gpsimenos, but @donatieng found a major flaw with our existing proposal. To make `#include events/EventQueue.h` work, we would need to set the top of the entire library as the include path 🤦  - `events/` is a subdirectory from the top.
We'll update the proposal soon.

**Review by LDong-Arm — CHANGES_REQUESTED** [review]
The directory structure looks good to me, but unit tests don't compile.
Please check `events`'s `unittest.cmake` and `UNITTESTS/CMakeList.txt`. I think include paths in the latter are global to all tests.

File: .astyleignore [hunk 1]
```diff
@@ -20,6 +20,8 @@
 ^features/unsupported/
 ^hal/storage_abstraction
 ^platform/cxxsupport
+^platform/test
+^events/test
```
Comment (evedon) [inline .astyleignore]: Why do you need to ignore the test directories?

File: events/test/unit/.mbedignore [hunk 2]
```diff
@@ -0,0 +1 @@
+*
```
Comment (evedon) [inline events/test/unit/.mbedignore]: Perhaps you could combine the different .mbedignore files.

**Review by evedon — CHANGES_REQUESTED** [review]
Note that the test directory is named `tests` not `test`. So it should be `events/tests`

File: .astyleignore (hunk 1 above)
Comment (gpsimenos) [reply .astyleignore]: Astyle wouldn't pass otherwise. I don't remember the exact failure.

Comment (evedon) [reply .astyleignore]: I meant that unless the test directories were previously ignored, we should not ignore them now. How many failures are there?

Comment (gpsimenos) [reply .astyleignore]: The UNITTESTS directory was previously ignored. Updated .astyleignore to only ignore UNITTESTS, not the entire tests directory.

File: events/test/unit/.mbedignore (hunk 2 above)
Comment (gpsimenos) [reply events/test/unit/.mbedignore]: Done.

**Comment by gpsimenos** [comment]
Unit tests now building.

File: .astyleignore [hunk 3]
```diff
@@ -21,6 +21,8 @@
 ^features/unsupported/
 ^hal/storage_abstraction
 ^platform/cxxsupport
+^platform/test
```
Comment (LDong-Arm) [inline .astyleignore]: ~Why is platform added?~ This path doesn't exist

**Review by LDong-Arm — APPROVED** [review]
LGTM apart from one question

File: .astyleignore (hunk 3 above)
Comment (gpsimenos) [reply .astyleignore]: Fixed.

**Review by LDong-Arm — APPROVED** [review]

**Comment by gpsimenos** [comment]
Rebased.

**Comment by evedon** [comment]
Could you manually run K64F::GCC_ARM::EVENTS-TESTS-TESTS-EVENTS-EQUEUE by turning the appropriate flag?
Everything else is good.

**Comment by gpsimenos** [comment]
Done. Results in description.

**Pull request merged**